elements/bulbs-video/components/revealed.js

```diff
@@ -156,30 +165,97 @@ export default class Revealed extends React.Component {
+    let prof;
+    if (window.isMobile.any) {
+      prof = 'theonion_mobileweb_html5';
+    }
+    else {
+      prof = 'theonion_desktop_html5';
+    }
+    return prof;
+  }
 
-    // AD_TYPE: one of p (preroll), m (midroll), po (postroll), o (overlay)
-    baseUrl += '&tt=p';
-    videoMeta.tags.push('html5'); // Force HTML 5
-    // Tags
-    baseUrl += '&t=' + videoMeta.tags;
-    //Category
-    let hostChannel = videoMeta.hostChannel;
-    let channel = videoMeta.channel_slug;
-    let series = videoMeta.series_slug;
-    let category = `${hostChannel}/${channel}`;
-    if (series) {
-      category += `/${series}`;
+  setDeviceAcronym () {
+    let deviceAcronym;
+    if (window.isMobile.any) {
+      deviceAcronym = 'm';
+    }
+    else {
+      deviceAcronym = 'd';
     }
-    baseUrl += '&s=' + category;
-    baseUrl += '&rnd=' + this.cacheBuster();
+    return deviceAcronym;
+  }
+
+  getSiteName () {
+    let targetChannel = this.props.video.targeting.target_channel;
+    return targetChannel.replace(/-/g, '');
+  }
 
-    if (vastTestId) {
-      baseUrl += '&xgid=' + vastTestId;
+  getDfpSection () {
+    let dfpSection;
+    if (window.TARGETING.dfp_section) {
+      dfpSection = window.TARGETING.dfp_section;
+    }
+    else if (window.TARGETING.dfp_specialcoverage) {
+      let slug = window.TARGETING.dfp_specialcoverage;
+      dfpSection = `specialcoverage_${slug}`;
     }
+    else {
+      dfpSection = 'video';
+    }
+    return dfpSection;
+  }
+
+  buildCsid (hostChannel) {
+    // Custom Site Section ID
+    // format: <device acronym>.<site name>_<dfp section>_<host channel>
+    let deviceAcronym = this.setDeviceAcronym();
+    let siteName = this.getSiteName();
+    let siteSection = this.getDfpSection();
+
+    return `${deviceAcronym}.${siteName}_${siteSection}_${hostChannel}`;
+  }
+
+  buildCaid (videohubReferenceId) {
+    // Custom content video asset id
+    // format: onion_<videohub reference id>
+    return `onion_${videohubReferenceId}`;
+  }
+
+  vastUrl (videoMeta) {
+    let hostChannel = videoMeta.hostChannel;
+    let videohubReferenceId = videoMeta.id;
+    let randomVideoPlayerNumber = videoMeta.vprn;
+    let vastTestId = this.vastTest(window.location.search);
+    let series = videoMeta.series_slug;
+    let campaignId = this.props.targetCampaignId;
+    let specialCoverage = this.props.targetSpecialCoverage;
+
+    let baseUrl = `http://${window.FREEWHEEL_NETWORK_ID}.v.fwmrm.net/ad/g/1?`;
+
+    // required global params
+    baseUrl += 'resp=' + 'vmap1';
+    baseUrl += '&prof=' + this.getProfValue();
+    baseUrl += '&csid=' + this.buildCsid(hostChannel);
+    baseUrl += '&caid=' + this.buildCaid(videohubReferenceId);
+    baseUrl += '&pvrn=' + this.cacheBuster();
+    baseUrl += '&vprn=' + randomVideoPlayerNumber;
+
+    // optional global param
+    if (vastTestId) { baseUrl += '&cana=' + vastTestId; }
+
+    // Key Values
+    baseUrl += ';&video_id=' + videohubReferenceId;
```

Comment:
We shouldn't need the ampersand after the semi-colon, so `;&video_id` becomes `;video_id`